doc/how-search-works.md

```diff
@@ -0,0 +1,50 @@
+# How Search Works
+
+Search API provides a `/search` endpoint for users to of the API to search
+for documents.
+
+You can also use `/batch_search` to send multiple queries.
+
+## Document retrieval
+
+On receiving a request to `/search` (a search query), Search API will parse
+the query, construct an Elasticsearch query, and then retrieve documents
+from Elasticsearch.
```

Comment:
Maybe add a little motivation here, something like "we do this so that other applications in the GOV.UK stack don't need to know the details of Elasticsearch"